File: Python/1637.py
# 1637. Widest Vertical Area Between Two Points Containing No Points
# https://leetcode.com/problems/widest-vertical-area-between-two-points-containing-no-points/

class Solution:
  def maxWidthOfVerticalArea(self, points: List[List[int]]) -> int:
    arr = list(map(lambda a: a[0], points))
    arr.sort()
    _max = 0

    for i in range(1, len(arr)):
      if (arr[i] == arr[i - 1]):
        continue

      _max = max(_max, arr[i] - arr[i - 1])

    return _max

File: Rust/1637.rs
/**
 * 1637. Widest Vertical Area Between Two Points Containing No Points
 * https://leetcode.com/problems/widest-vertical-area-between-two-points-containing-no-points/
 */

impl Solution {
  pub fn max_width_of_vertical_area(points: Vec<Vec<i32>>) -> i32 {
    let mut arr: Vec<i32> = points
      .into_iter()
      .map(|a| a[0])
      .collect();
    arr.sort();
    let mut max = 0;

    for i in 1..arr.len() {
      if arr[i] == arr[i - 1] {
        continue;
      }

      max = std::cmp::max(max, arr[i] - arr[i - 1]);
    }

    return max;
  }
}
